### PR TITLE
Fixing parsing of Go Test output

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

It turns out that `go test --json` doesn't exclusively product JSON output - sometimes it outputs plain text lines as well:

```
{"Time":"2022-05-14T13:30:13.483275253+12:00","Action":"output","Package":"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel","Output":"ok  \tgithub.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel\t4.986s\tcoverage: 13.1% of statements in ./...\n"}
{"Time":"2022-05-14T13:30:13.483361877+12:00","Action":"pass","Package":"github.com/Azure/azure-service-operator/v2/tools/generator/internal/astmodel","Elapsed":4.989}
FAIL	github.com/Azure/azure-service-operator/v2/tools/generator/internal/codegen/pipeline [build failed]
{"Time":"2022-05-14T13:30:13.48514437+12:00","Action":"output","Package":"github.com/Azure/azure-service-operator/v2/tools/generator/internal/codegen/pipeline/recursivetypefixer","Output":"?   \tgithub.com/Azure/azure-service-operator/v2/tools/generator/internal/codegen/pipeline/recursivetypefixer\t[no test files]\n"}
{"Time":"2022-05-14T13:30:13.48517613+12:00","Action":"skip","Package":"github.com/Azure/azure-service-operator/v2/tools/generator/internal/codegen/pipeline/recursivetypefixer","Elapsed":0}
```

To handle this, I modified the test reporter to parse the log line by line and to skip expected text lines.

**How does this PR make you feel**:
gif

### CODE REVIEW

File: v2/tools/mangle-test-json/main.go [hunk 1]
```diff
@@ -78,16 +81,33 @@ func actionSymbol(d TestRun) string {
...
-	data := []JSONFormat{}
-	err = json.Unmarshal([]byte(jsonData), &data)
-	if err != nil {
-		log.Fatalf("%e", err)
+	var data []JSONFormat
+	errCount := 0
+	for row, line := range lines {
+		// Skip empty lines and lines starting with "FAIL"
+		if line == "" ||
+			strings.HasPrefix(line, "FAIL") {
```
Comment (Porges) [inline v2/tools/mangle-test-json/main.go:95]: The line should be printed to `stderr` as well.

File: v2/tools/mangle-test-json/main.go [hunk 2]
```diff
@@ -34,6 +35,7 @@ type TestRun struct {
 
 func main() {
 	for _, testOutputFile := range os.Args[1:] {
+		log.Printf("Parsing  %s\n\n", testOutputFile)
```
Comment (Porges) [inline v2/tools/mangle-test-json/main.go:38]: Does this go to `stderr?`

Comment (Porges) [reply v2/tools/mangle-test-json/main.go:38]: Sounds like yes.

**Review by Porges — CHANGES_REQUESTED** [review]
`FAIL` lines need to go to `stderr` as well or they won't be visible.

File: v2/tools/mangle-test-json/main.go (hunk 2 above)
Comment (theunrepentantgeek) [reply v2/tools/mangle-test-json/main.go:38]: Yes - appears in the action log, not in the markdown file:

```
[produce-markdown-summary] 2022/05/15 20:49:21 Parsing  /workspace/reports/controller-integration-tests.json
[produce-markdown-summary] 
[produce-markdown-summary] 2022/05/15 20:49:21 Complete. 
```

**Review by Porges — APPROVED** [review]

**Pull request merged**